org.sil.scripture-app-builder.yml

```diff
@@ -0,0 +1,86 @@
+  # Node20:    https://github.com/flathub/org.freedesktop.Sdk.Extension.node20/blob/branch/23.08/org.freedesktop.Sdk.Extension.node20.yaml
+app-id: org.sil.scripture-app-builder
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.openjdk17
+  # - org.freedesktop.Sdk.Extension.openjdk21
+finish-args:
+  # Window x11 socket access
+  - --socket=x11
+  # Audio access
+  - --socket=pulseaudio
+  # Network access
+  - --share=network
+  # XShm access
+  - --share=ipc
+  # All external devices (catch all for devices we interact with and graphical acceleration)
+  - --device=all
+  # Local file browsing access
+  - --filesystem=host
+  # Temporary file access
+  - --filesystem=/tmp
+  # USB Drives
+  - --system-talk-name=org.freedesktop.Udisks2
+  # File saving access
+  - --talk-name=org.freedesktop.FileManager1
+
+# App's run command
+command: sab-run.sh
+
+modules:
+
+# For in-app terminal
+- xterm.yml
+
+- python3-aeneas.yml
+
+# For python scripts
+- python3-requests.yaml
+
+# For pwa
+- sab-node20.yml
+
+- openjdk17.yml
+
+- name: scripture-app-builder
+  buildsystem: simple
+  build-commands:
+    # Source tarball from a docker image and extract to flatpak app
+    - mkdir app-builder
+    - tar xf scripture-app-builder.tgz -C app-builder
+    # Copy only the necessary files
+    - mkdir -p /app/scripture-app-builder/bin
+    - cp -RPn app-builder/bin/* /app/scripture-app-builder/bin/
+    - cp -RPn app-builder/{fonts,images,info,lib,docs,tools,pwa} /app/scripture-app-builder/
+    # Icons, desktop entry, metainfo and runner script
+    - install -Dm644 app-builder/app-icons/scripture-app-builder.png /app/share/icons/hicolor/128x128/apps/org.sil.scripture-app-builder.png
+    - install -Dm644 app-builder/app-icons/scripture-app-builder.svg /app/share/icons/hicolor/scalable/apps/org.sil.scripture-app-builder.svg
+    - desktop-file-install --dir /app/share/applications app-builder/flatpak/org.sil.scripture-app-builder.desktop
+    - install -Dm644 app-builder/flatpak/org.sil.scripture-app-builder.metainfo.xml -t /app/share/metainfo/
+    - install -Dm755 sab-run.sh /app/bin/sab-run.sh
+    # JavaFX SDK
+    - unzip -q -j -d sdk sdk.zip '*lib/*'
+    - mv sdk /app/lib/sdk
+  cleanup:
+    - '/app-builder'
+  sources:
+    - type: file
+      path: sab-run.sh
+    # JavaFX .zip archive
+    - type: file
+      only-arches:
+      - x86_64
+      dest-filename: sdk.zip
+      url: https://download2.gluonhq.com/openjfx/17.0.11/openjfx-17.0.11_linux-x64_bin-sdk.zip
+      sha256: f46a1fbea32b83cca6715d74cba7d9c24ce320f1da2daf8ff852f133e9f15674
+    # App Builder prebuilt binaries
+    - type: file
+      only-arches:
+      - x86_64
+      url: https://sil-app-builders-linux-artifacts.s3.amazonaws.com/artifacts/59/scripture-app-builder.tgz
+      sha256: 1689bf31291802614401042f5c263c1c30d10a546982570ada7ac2d6924e77ff 
```

Comment:
This is unversioned ? It will break whenever the source is rotated because sha256 will not match anymore.

Reply:
It is versioned by the build number (59).  Every time we do a release, I will update the url and the sha256 code.